openpathsampling/random.py

```diff
@@ -0,0 +1,25 @@
+from __future__ import absolute_import
+import numpy as np
+import random
+
+
+# Set one rng for all of OPS
+if np.version.version < '1.17':
+    # Legacy support, remove when numpy 1.16 support is dropped
+    class RandomState(np.random.RandomState):
+        def __init__(self, *args, **kwargs):
+            super(RandomState, self).__init__(*args, **kwargs)
+
+        # method overrides to mimic numpy.random.Generator
+        def random(self, *args, **kwargs):
+            return self.random_sample(*args, **kwargs)
+
+        def integers(self, *args, **kwargs):
+            return self.randint(*args, **kwargs)
+    rng = RandomState()
+else:
+    rng = np.random.default_rng()
+
+
+def default_rng():
```

Comment:
I don't think this should be a function. Is there a specific reason it must be? Here are my reasons:

1. A function implies to users that this is creating something.
2. The behavior is specifically different from what `np.random.default_rng()` does.

If you do your `TestRandom.test_identities` test with `np.random.default_rng()`, it will fail the `is` test.

Instead, either name `rng` as `default_rng` or `DEFAULT_RNG` (to really emphasize that it is constant). You can still import it into the main `openpathsampling` namespace.

Reply:
nvm. I think this has to be a function so that the underlying `rng` object can change, but newly created instances get the new `rng`. Nonetheless, please rename `rng` to something like `DEFAULT_RNG`.

If I understand correctly, in that case, setting the seed before a simulation starts would be something like `paths.random.DEFAULT_RNG = np.random.default_rng(seed)` before creating any objects. Is that right?

Reply:
> Nonetheless, please rename rng to something like DEFAULT_RNG

Will do

> If I understand correctly, in that case, setting the seed before a simulation starts would be something like paths.random.DEFAULT_RNG = np.random.default_rng(seed) before creating any objects. Is that right?

Yes, exactly (except that it has to go through `paths.random.RandomState(seed)` for `np<1.17`) .